### PR TITLE
Move topology monitoring data fetching into Hooks

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**Fixes**
https://issues.redhat.com/browse/ODC-4987

**Description**
Create hooks for retrieving prometheus rules and overview alerts and metrics.

**Browser conformance**: 
- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge

/kind cleanup

### CODE REVIEW

File: frontend/packages/dev-console/src/components/topology/TopologyDataRetriever.tsx [hunk 1]
```diff
@@ -38,20 +34,27 @@ export const ConnectedTopologyDataRetriever: React.FC<TopologyDataRetrieverProps
...
+      // Don't update due to time changes
+      alertData.forEach((alert) => {
+        delete alert.activeAt;
+        if (alert.rule) {
+          delete alert.rule.evaluationTime;
+          delete alert.rule.lastEvaluation;
+          alert.rule.alerts &&
+            alert.rule.alerts.forEach((ruleAlert) => {
+              delete ruleAlert.activeAt;
+            });
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/topology/TopologyDataRetriever.tsx:52]:
```suggestion
          alert.rule.alerts?.forEach((ruleAlert) => {
            delete ruleAlert.activeAt;
          });
```

File: frontend/packages/dev-console/src/components/topology/hooks/useOverviewAlertsUpdater.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      })
+      .catch((e) => {
+        console.error(e); // eslint-disable-line no-console
+      });
+  };
+
+  fetchAlerts();
+
+  return () => {
+    clearInterval(alertsInterval);
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/topology/hooks/useOverviewAlertsUpdater.ts:32]: I don't tested it clearInterval could also clear a timeout, but I think you could also use clearTimeout here
```suggestion
    clearTimeout(timeout);
```

File: frontend/packages/dev-console/src/components/topology/hooks/useOverviewMetricsUpdater.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      })
+      .catch((e) => {
+        console.error(e); // eslint-disable-line no-console
+      });
+  };
+
+  fetchMetrics();
+
+  return () => {
+    clearInterval(metricsInterval);
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/topology/hooks/useOverviewMetricsUpdater.ts:48]: Same question here. It's a timeout actually?

File: frontend/public/components/graphs/prometheus-rules-hook.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { useURLPoll } from '../utils/url-poll-hook';
+import { getPrometheusURL, PrometheusEndpoint } from './helpers';
+import { PrometheusRulesResponse } from '../monitoring/types';
+
+export const usePrometheusRulesPoll = ({ delay, namespace }: PrometheusPollProps) => {
+  const url = getPrometheusURL({
+    endpoint: PrometheusEndpoint.RULES,
+    namespace,
+    query: `namespace=${namespace}`,
+  });
```
Comment (jerolimov) [inline frontend/public/components/graphs/prometheus-rules-hook.ts:10]: Here I have a just question, is this change the expected behaviour? I double checked the API calls to the backend, and the query parameter is new here:

Old: `/api/prometheus-tenancy/api/v1/rules?namespace=christoph-test`
New: `/api/prometheus-tenancy/api/v1/rules?namespace=christoph-test&query=namespace%3Dchristoph-test`

Is this "okay" or unnecessary?

File: frontend/packages/dev-console/src/components/topology/TopologyDataRetriever.tsx (hunk 1 above)
Comment (jeff-phillips-18) [reply frontend/packages/dev-console/src/components/topology/TopologyDataRetriever.tsx:52]: eslint doesn't like that

File: frontend/packages/dev-console/src/components/topology/hooks/useOverviewAlertsUpdater.ts (hunk 2 above)
Comment (jeff-phillips-18) [reply frontend/packages/dev-console/src/components/topology/hooks/useOverviewAlertsUpdater.ts:32]: Correct. If the update fails we still want to attempt to fetch again.

File: frontend/public/components/graphs/prometheus-rules-hook.ts (hunk 4 above)
Comment (jeff-phillips-18) [reply frontend/public/components/graphs/prometheus-rules-hook.ts:10]: Right, unnecessary `query` prop in `metricUtils.ts`

**Comment by jeff-phillips-18** [comment]
Updated. @jerolimov PTAL

File: frontend/packages/dev-console/src/components/monitoring/alerts/MonitoringAlerts.tsx [hunk 5]
```diff
@@ -24,6 +22,7 @@ import {
   applyListSort,
 } from './monitoring-alerts-utils';
 import './MonitoringAlerts.scss';
+import { usePrometheusRulesPoll } from '@console/internal/components/graphs/prometheus-rules-hook';
```
Comment (vikram-raj) [inline frontend/packages/dev-console/src/components/monitoring/alerts/MonitoringAlerts.tsx:25]: nit: sort imports.

**Review by vikram-raj — COMMENTED** [review]
Verified and tested the Monitoring Alerts and Topology Alerts and it works as expected.

**Review by rohitkrai03 — APPROVED** [review]
/lgtm

**Pull request merged**